aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -297,19 +298,33 @@ pub struct AggregatorData {
 
 impl AggregatorData {
     /// Returns a mutable reference to an aggregator with `id` and a `limit`.
-    /// If transaction that is currently executing did not initilize it), a new
-    /// aggregator instance is created, with a zero-initialized value and in a
-    /// delta state.
+    /// If transaction that is currently executing did not initilize it, a new aggregator instance is created.
+    /// The state of the new aggregator instance depends on the `aggregator_enabled` flag.
+    /// If the `aggregator_enabled` flag is true, the new aggregator instance
+    /// is initialized with zero and in a delta state.
+    /// If the `aggregator_enabled` flag is false, the new aggregator instance
+    /// is initialized in the Data state with its latest value.
     /// Note: when we say "aggregator instance" here we refer to Rust struct and
     /// not to the Move aggregator.
-    pub fn get_aggregator(&mut self, id: AggregatorID, limit: u128) -> &mut Aggregator {
+    pub fn get_aggregator(
+        &mut self,
+        id: AggregatorID,
+        limit: u128,
+        resolver: &dyn TableResolver,
+        aggregator_enabled: bool,
+    ) -> PartialVMResult<&mut Aggregator> {
         self.aggregators.entry(id).or_insert_with(|| Aggregator {
```

Comment:
you might be able to do:
`let aggregator = self.aggregators.entry(id).or_insert(Aggregator {...});`
Basically you should be able to extract mutable reference from entry here itself, instead of calling get_mut again.
I also think or_insert should work instead of or_insert_with but lint would know better, so maybe not.
(I do ./scripts/rust_lint.sh --check in aptos-core, but I think github flows also run it)